public/app/plugins/datasource/influxdb/components/useShadowedState.ts

```diff
@@ -5,9 +5,11 @@ export function useShadowedState<T>(outsideVal: T): [T, (newVal: T) => void] {
   const prevOutsideVal = useRef(outsideVal);
 
   useEffect(() => {
+    const isOutsideValChanged = prevOutsideVal.current !== outsideVal;
```

Comment:
Sorry for not noticing this but we have a react-hooks lib already in deps that already has `usePrev` hook that can be used for this.